### PR TITLE
msconvert: add demultiplex filter

### PR DESCRIPTION
**Pull request by chambm**

### CODE REVIEW

**Comment by bgruening** [comment]
@chambm please bump the version number.

**Comment by chambm** [comment]
@bgruening This looks like a failure running docker rather than a tool or wrapper failure.

**Comment by bernt-matthias** [comment]
Is the test running locally for you? With the github actions testing we switched to `planemo test --biocontainers --no_conda_auto_init`  before it was `planemo test --docker` (there was a special case for msconvert). But since ` --biocontainers` implies `--docker` it should be OK.

**Comment by bernt-matthias** [comment]
Container resolution seems to work: `Checking with container resolver [ExplicitContainerResolver[]] found description [ContainerDescription[identifier=chambm/pwiz-skyline-i-agree-to-the-vendor-licenses:3.0.19052-089e81090,type=docker]]`

**Comment by chambm** [comment]
I haven't tried it locally, but the error is `sudo: unknown uid 1001: who are you?`. I'm not sure which part of the command is issuing that error though. Nor why the change you mentioned would cause the error now but not the last time these tests ran.

File: tools/msconvert/msconvert_macros.xml [hunk 1]
```diff
@@ -376,6 +379,30 @@
 
       <param name="ms2deisotope" type="boolean" label="Deisotope MS2 using Markey method" help="" truevalue="true" falsevalue="false" />
 
+      <conditional name="demultiplex">
```
[the remark this answers is not in the document]
Comment (chambm) [reply tools/msconvert/msconvert_macros.xml:382]: Practically impossible to do with a reasonable size AFAIK. We don't have a unit test for it in pwiz either.

[the remark this answers is not in the document]
Comment (chambm) [reply tools/msconvert/msconvert_macros.xml:382]: Well actually we do but it's generate procedurally by the test itself.

**Comment by bgruening** [comment]
@chambm do you remember that we added those hacks: https://github.com/galaxyproteomics/tools-galaxyp/pull/470/files#diff-2016f22d4f8ce3a3a8566735439b96a8R42

Have you changed anything in your container?

**Comment by chambm** [comment]
@bgruening That link takes me to the diff with this PR's changes. There haven't been significant changes in the pwiz container. Mostly just adding users for galaxy and the mywine script which is an alternative to the wine64_anyuser script: https://github.com/ProteoWizard/container/commits/master/Dockerfile

**Comment by bgruening** [comment]
It passes locally!

**Comment by bgruening** [comment]
Thanks @chambm!

**Pull request merged**